### PR TITLE
feature/NEX-132_vendorization of item runner

### PR DESCRIPTION
**Pull request by btamas**
Vendorization of tao item runner.

Jira: https://oat-sa.atlassian.net/browse/NEX-132

Post task:
- [ ] release to npm registry

### CODE REVIEW

**Review by ferenckiss89 — APPROVED** [review]
Tested, all good.

File: src/assets/manager.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015-2017 (original work) Open Assessment Technologies SA;
```
Comment (oatymart) [inline src/assets/manager.js]: Does the year not need to become 2019 as we are editing these files (even if their functionality is unchanged) in 2019?

File: src/assets/manager.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+         */
+        prependStrategy: function prependStrategy(strategy) {
+            checkStrategy(strategy);
+
+            this._strategies.unshift(strategy);
+        },
+
+        /**
+         * Change the strategies
+         * @param {AssetStrategy[]} strategies - the strategies
```
Comment (oatymart) [inline src/assets/manager.js:134]: The documented param name doesn't match the one used in the function.

File: src/assets/manager.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    strategies = _.isArray(strategies) ? strategies : [strategies];
+    data = data || {};
+    options = _.defaults(options || {}, {
+        parseUrl: true
+    });
+
+    /**
+     * A brand new asset manager is created by the factory
+     */
+    var assetManager = {
```
Comment (oatymart) [inline src/assets/manager.js]: Can we define `var assetManager` at the top, for ESLint & file readability?

File: src/runner/api/itemRunner.js [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+ * @param {String} [providerName] - the name of a provider previously registered see {@link itemRunnerFactory#register}
+ * @param {Object} [data] - the data of the item to run
+ *
+ * @returns {itemRunner}
+ */
+var itemRunnerFactory = function itemRunnerFactory(providerName, data, options) {
+    var provider, providers, assetManager;
+
+    //contains the bound events.
+    var events = {};
```
Comment (oatymart) [inline src/runner/api/itemRunner.js]: Is it used? Can we safely remove it?

File: src/runner/api/itemRunner.js [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+         * @fires itemRunner#clear
+         */
+        clear: function() {
+            var self = this;
+
+            /**
+             * Call back when clear is done
+             */
+            var clearDone = function clearDone() {
+                //events = {};
```
Comment (oatymart) [inline src/runner/api/itemRunner.js]: I guess this was the purpose for `events`. But it seems to be abandoned code now.

**Review by oatymart — CHANGES_REQUESTED** [review]
A question about copyright years, and I just highlighted a couple of things the linter highlighted which we could take the time to improve.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
+{
+    "name": "@oat-sa/tao-item-runner",
+    "version": "0.1.0",
+    "displayName": "TAO Item Runner",
```
Comment (krampstudio) [inline package.json:4]: Could you please add the `description` field. Without it `npm install` will display a warning

File: src/runner/README.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
+# Runner quick overview
```
Comment (krampstudio) [inline src/runner/README.md:1]: Could you please at least link this readme from the main readme ?

**Review by krampstudio — CHANGES_REQUESTED** [review]

**Review by krampstudio — APPROVED** [review]

**Pull request merged**